src/java/org/apache/cassandra/cql3/statements/schema/CreateTableStatement.java

```diff
@@ -104,34 +113,77 @@ public Keyspaces apply(ClusterMetadata metadata)
+
+            KeyspaceMetadata likeKeyspaceMeta = schema.getNullable(likeKeyspaceName);
+            if (null == likeKeyspaceMeta)
+                throw ire("Target Keyspace '%s' doesn't exist", likeKeyspaceName);
+
+            targetKeyspace = likeKeyspaceMeta;
+            if (likeKeyspaceMeta.hasTable(likeTableName))
+            {
+                throw new AlreadyExistsException(likeKeyspaceName, likeTableName);
+            }
 
-            throw new AlreadyExistsException(keyspaceName, tableName);
+            // todo support udt for differenet ks latter
+            if (!keyspaceName.equalsIgnoreCase(likeKeyspaceName) &&
+                !keyspace.types.isEmpty())
+            {
+                throw ire("Cannot use CTREAT TABLE LIKE across different keyspace when source table have UDT.");
+            }
+
+            table = buildTargetTableMeta(metadata, sourceTableMeta);
         }
+        else
+        {
+            assert likeKeyspaceName == null && likeTableName == null ;
+            if (keyspace.hasTable(tableName))
+            {
+                if (ifNotExists)
+                    return schema;
 
-        // add all user functions to be able to give a good error message to the user if the alter references
-        // a function from another keyspace
-        UserFunctions.Builder ufBuilder = UserFunctions.builder().add();
-        for (KeyspaceMetadata ksm : schema)
-            ufBuilder.add(ksm.userFunctions);
+                throw new AlreadyExistsException(keyspaceName, tableName);
+            }
 
-        TableMetadata.Builder builder = builder(keyspace.types, ufBuilder.build()).epoch(metadata.nextEpoch());
+            // add all user functions to be able to give a good error message to the user if the alter references
+            // a function from another keyspace
+            UserFunctions.Builder ufBuilder = UserFunctions.builder().add();
+            for (KeyspaceMetadata ksm : schema)
+                ufBuilder.add(ksm.userFunctions);
 
-        // We do not want to set table ID here just yet, since we are using CQL for serialising a fully expanded CREATE TABLE statement.
-        this.expandedCql = builder.build().toCqlString(false, attrs.hasProperty(TableAttributes.ID), ifNotExists);
+            TableMetadata.Builder builder = builder(keyspace.types, ufBuilder.build()).epoch(metadata.nextEpoch());
 
-        if (!attrs.hasProperty(TableAttributes.ID))
-            builder.id(TableId.get(metadata));
-        TableMetadata table = builder.build();
-        table.validate();
+            // We do not want to set table ID here just yet, since we are using CQL for serialising a fully expanded CREATE TABLE statement.
+            this.expandedCql = builder.build().toCqlString(false, attrs.hasProperty(TableAttributes.ID), ifNotExists);
 
-        if (keyspace.replicationStrategy.hasTransientReplicas()
+            if (!attrs.hasProperty(TableAttributes.ID))
+                builder.id(TableId.get(metadata));
```

Comment:
I am not sure we need this, id should be automatically set when `build` is called.